server/app/api/v1/charity/raffle/controller.js

```diff
@@ -153,7 +154,7 @@ const purchase = (req, res, next) => {
   function addTickets(user) {
     // checks for if user already purchased raffles
     const purchaseIndex = findIndex(req.raffle.purchases, ['buyer', user._id]);
-    const userRafflesIndex = findIndex(user.raffles, req.raffle._id);
+    const userRaffles = find(user.raffles, req.raffle._id);
```

Comment:
See lines 161, 162 for more info on why I needed to keep `findIndex` for `purchase`